config/initializers/new_framework_defaults_7_1.rb

```diff
@@ -1,214 +0,0 @@
-# frozen_string_literal: true
-
-# Be sure to restart your server when you modify this file.
-#
-# This file eases your Rails 7.1 framework defaults upgrade.
-#
-# Uncomment each configuration one by one to switch to the new default.
-# Once your application is ready to run with all new defaults, you can remove
-# this file and set the `config.load_defaults` to `7.1`.
-#
-# Read the Guide for Upgrading Ruby on Rails for more info on each option.
-# https://guides.rubyonrails.org/upgrading_ruby_on_rails.html
-
-# No longer add autoloaded paths into `$LOAD_PATH`. This means that you won't be able
-# to manually require files that are managed by the autoloader, which you shouldn't do anyway.
-# This will reduce the size of the load path, making `require` faster if you don't use bootsnap, or reduce the size
-# of the bootsnap cache if you use it.
-Rails.application.config.add_autoload_paths_to_load_path = false
-
-# Remove the default X-Download-Options headers since it is used only by Internet Explorer.
-# If you need to support Internet Explorer, add back `"X-Download-Options" => "noopen"`.
-# Rails.application.config.action_dispatch.default_headers = {
-#   "X-Frame-Options" => "SAMEORIGIN",
-#   "X-XSS-Protection" => "0",
-#   "X-Content-Type-Options" => "nosniff",
-#   "X-Permitted-Cross-Domain-Policies" => "none",
-#   "Referrer-Policy" => "strict-origin-when-cross-origin"
-# }
-
-# Do not treat an `ActionController::Parameters` instance
-# as equal to an equivalent `Hash` by default.
-Rails.application.config.action_controller.allow_deprecated_parameters_hash_equality = false
-
-# Active Record Encryption now uses SHA-256 as its hash digest algorithm. Important: If you have
-# data encrypted with previous Rails versions, there are two scenarios to consider:
-#
-# 1. If you have +config.active_support.key_generator_hash_digest_class+ configured as SHA1 (the default
-# before Rails 7.0), you need to configure SHA-1 for Active Record Encryption too:
-# Rails.application.config.active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA1
-# 2. If you have +config.active_support.key_generator_hash_digest_class+ configured as SHA256 (the new default
-# in 7.0), then you need to configure SHA-256 for Active Record Encryption:
-# Rails.application.config.active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA256
-#
-# If you don't currently have data encrypted with Active Record encryption, you can disable this setting to
```

Comment:
I think we are effectively in this scenario -- we have the recently enabled otp AR encryption column, but that has only existed on 7.1 and has never been in a release, so I think we are in the SHA256 case described above, and having the framework default of not backwards-supporting SHA1 is acceptable.

Reply:
I would have thought that as well, but:
```
May 31 14:59:06 mastodon bundle[993665]: E, [2024-05-31T14:59:06.438057 #993665] ERROR -- : [64fdf096-f096-476a-a51b-9cc20e91919a]
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a] ActiveRecord::Encryption::Errors::Decryption (ActiveRecord::Encryption::Errors::Decryption):
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a]
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a] app/controllers/concerns/auth/two_factor_authentication_concern.rb:32:in `valid_otp_attempt?'
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a] app/controllers/concerns/auth/two_factor_authentication_concern.rb:74:in `authenticate_with_two_factor_via_otp'
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a] app/controllers/concerns/auth/two_factor_authentication_concern.rb:51:in `authenticate_with_two_factor'
May 31 14:59:06 mastodon bundle[993665]: [64fdf096-f096-476a-a51b-9cc20e91919a] lib/mastodon/rack_middleware.rb:9:in `call'
```

I'm not completely sure this is because of this particular setting, though.